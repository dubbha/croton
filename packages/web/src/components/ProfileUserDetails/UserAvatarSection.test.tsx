import React from 'react';
import { render } from '@testing-library/react';

describe('components/UserAvatarSection', () => {
  const firstName = 'Chuck';
  const lastName = 'Norris';
  const pictureUrl = 'https://my-pics/selfie-with-big-bang.png';
  const facebookId = '0';
  const mockSocialProfile = { pictureUrl, facebookId };

  const setUpMocks = ({ socialProfile }) => {
    const mockUseSelector = jest.fn().mockReturnValue({
      socialProfile,
      firstName,
      lastName,
    });

    jest.doMock('react-redux', () => ({
      useSelector: mockUseSelector,
    }));

    const { UserAvatarSection } = require('./UserAvatarSection');
    const { container, getByTestId } = render(<UserAvatarSection />);

    return {
      container,
      mockUseSelector,
      getByTestId,
    };
  };

  afterEach(() => {
    jest.resetAllMocks();
  });

  it('should render successfully', () => {
    jest.isolateModules(() => {
      const { container } = setUpMocks({
        socialProfile: mockSocialProfile,
      });

      expect(container.firstChild).toMatchSnapshot();
    });
  });

  it('should pick up picture url from the social profile and set it up as a header', () => {
    jest.isolateModules(() => {
      const { getByTestId } = setUpMocks({
        socialProfile: mockSocialProfile,
      });

      const pictureContainer = getByTestId('userAvatarSection__picture');

      expect(pictureContainer.getElementsByTagName('source')[0]['src']).toBe(
        pictureUrl
      );
      expect(pictureContainer.getElementsByTagName('img')[0]['src']).toBe(
        pictureUrl
      );
    });
  });

  it('should pick up picture url from the social profile and set it up as a header', () => {
    jest.isolateModules(() => {
      const { getByTestId } = setUpMocks({
        socialProfile: undefined,
      });

      expect(getByTestId('userAvatarSection__default')).toBeTruthy();
    });
  });
});
