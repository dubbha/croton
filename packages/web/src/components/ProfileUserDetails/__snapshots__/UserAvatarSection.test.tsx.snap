// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/UserAvatarSection should render successfully 1`] = `
<picture
  data-testid="userAvatarSection__picture"
>
  <source
    class="circle with-picture"
    src="https://my-pics/selfie-with-big-bang.png"
  />
  <img
    alt="C N"
    class="circle with-picture"
    src="https://my-pics/selfie-with-big-bang.png"
  />
</picture>
`;
